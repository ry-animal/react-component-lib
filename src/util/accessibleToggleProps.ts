export function accessibleToggleProps(
  label: string,
  controls: string,
  expanded: boolean,
  popupType?: "menu"
) {
  return {
    "aria-controls": controls,
    "aria-expanded": expanded,
    "aria-haspopup": popupType || true,
    "aria-label": label,
    "aria-pressed": expanded,
    role: "button",
  };
}
