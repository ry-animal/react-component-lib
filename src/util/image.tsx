import { ISelectedFile } from "src/components/Forms/FileUpload/ISelectedFile";

export const makeDataUrl = (selectedFile: ISelectedFile) =>
  URL.createObjectURL(selectedFile.file);
