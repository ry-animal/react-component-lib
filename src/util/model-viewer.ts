/*
 * This file is a hacky way of allowing any other component to
 * use the <model-viewer /> tag just by doing:
 *   import "../../../util/model-viewer";
 *
 * The authors of the package do not expose it in a React-friendly
 * way, see: https://github.com/google/model-viewer/issues/1502
 *
 * The component documentation is at https://modelviewer.dev/docs/
 */

import "@google/model-viewer";

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace JSX {
    export interface IntrinsicElements {
      "model-viewer": {
        src?: string;
        alt?: string;
        poster?: string;
        autoplay?: boolean;
        exposure?: number;
        "camera-controls"?: boolean;
        "environment-image"?: string;
        "seamless-poster"?: string;
        loading?: "auto" | "lazy" | "eager";
        reveal?: "auto" | "interaction" | "manual";
      } & React.DetailedHTMLProps<
        React.HTMLAttributes<HTMLElement>,
        HTMLElement
      >;
    }
  }
}
