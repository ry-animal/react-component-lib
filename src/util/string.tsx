const DEFAULT_LOCALE = "en-US";

export const truncateEthAddress = (str: string): string => {
  if (str.length) {
    const numStartingCharacters = 6;
    const numTrailingCharacters = 4;
    return `${str.substring(0, numStartingCharacters)}...${str.substring(
      str.length - numTrailingCharacters,
      str.length
    )}`;
  }
  return "";
};

export const formatUSDPriceString = (value: number): string => {
  const minimumFractionDigits = 2;
  const maximumFractionDigits = 2;
  return `$${value.toLocaleString(DEFAULT_LOCALE, {
    minimumFractionDigits,
    maximumFractionDigits,
  })}`;
};

export const formatETHPriceString = (priceInEth: number): string => {
  const minimumFractionDigits = 0;
  const maximumFractionDigits = 8;
  const maxDigitsWithDecimal = 8;

  let formatted = `${priceInEth
    .toLocaleString(DEFAULT_LOCALE, {
      minimumFractionDigits,
      maximumFractionDigits,
    })
    .replace(/,/gu, "")
    .substring(0, maxDigitsWithDecimal)}`;

  if (formatted.endsWith(".")) {
    formatted = formatted.slice(0, -1);
  }
  return formatted;
};

const evaluateNumberLessThanOne = (num: number) => {
  if (num === 0) {
    return "0";
  }

  const minDecimalValue = 0.0001;
  if (num < minDecimalValue) {
    return minDecimalValue.toString();
  }
  // To round decimal to 4 digits
  const maxDigits = 10000;
  return `${Math.round(num * maxDigits) / maxDigits}`;
};

// Abbreviate Number https://stackoverflow.com/a/9462382

export const abbreviateNumber = (num: number): string => {
  const lookup = [
    { value: 1, symbol: "" },
    { value: 1e3, symbol: "k" },
    { value: 1e6, symbol: "M" },
    { value: 1e9, symbol: "B" },
    { value: 1e12, symbol: "T" },
    { value: 1e15, symbol: "P" },
    { value: 1e18, symbol: "E" },
  ];

  const rx = /\.0+$|(?<id>\.[0-9]*[1-9])0+$/u;
  const digits = 1;

  const item = lookup
    .slice()
    .reverse()
    .find((x) => num >= x.value);
  return item
    ? (num / item.value).toFixed(digits).replace(rx, "$1") + item.symbol
    : evaluateNumberLessThanOne(num);
};

export const getDomainFromRegex = (url: string) => {
  const matches = url.match(/:\/\/(?<id>.[^/]+)/u);
  // Extract domain (will be null if no match is found)
  return matches && matches[1];
};
