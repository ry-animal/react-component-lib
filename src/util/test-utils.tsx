import { render, RenderOptions } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { LIGHT_THEME } from "../theme";

interface ProvidersProps {
  children: React.ReactNode;
}

function Providers({ children }: ProvidersProps) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={LIGHT_THEME}>{children}</ThemeProvider>
    </BrowserRouter>
  );
}

const customRender = (
  ui: JSX.Element,
  options?:
    | RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement
      >
    | undefined
) => render(ui, { wrapper: Providers, ...options });

// Duplicated instead of import "../../__mocks__/matchMediaMock"; because jest has trouble resolving imports
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// Re-export everything
export * from "@testing-library/react";
// Override render method
export { customRender as render };
