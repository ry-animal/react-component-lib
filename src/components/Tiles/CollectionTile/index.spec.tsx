import userEvent from "@testing-library/user-event";
import { CollectionTile, CollectionTileLoader } from ".";
import { render, screen } from "../../../util/test-utils";

describe("<CollectionTile />", () => {
  const sampleProps = () => ({
    collectionImgSrc: "collection.jpg",
    collectionName: "Collection name",
    creatorImgSrc: "creator.png",
    creatorName: "Creator",
  });
  it("Default rendering matches snapshot", () => {
    expect(
      render(
        <CollectionTile {...sampleProps()} onClick={() => {}} />
      ).asFragment()
    ).toMatchSnapshot();
  });

  it("Default loading matches snapshot", () => {
    expect(render(<CollectionTileLoader />).asFragment()).toMatchSnapshot();
  });

  it("Default rendering matches snapshot (size = small)", () => {
    expect(
      render(
        <CollectionTile {...sampleProps()} onClick={() => {}} contentSize="s" />
      ).asFragment()
    ).toMatchSnapshot();
  });

  it("Default rendering matches snapshot (disabled)", () => {
    expect(
      render(
        <CollectionTile {...sampleProps()} onClick={() => {}} disabled={true} />
      ).asFragment()
    ).toMatchSnapshot();
  });

  it("Renders anchor if url provided", () => {
    const url = "/collection/url";
    render(<CollectionTile {...sampleProps()} collectionUrl={url} />);
    expect(screen.getByRole("link")).toHaveAttribute("href", url);
  });

  it("is Loopring", () => {
    render(<CollectionTile {...sampleProps()} layer="Loopring" />);
    expect(screen.getByTestId("icon-lr")).toHaveAttribute(
      "fill",
      "transparent"
    );
  });

  it("is L1", () => {
    render(<CollectionTile {...sampleProps()} layer="L1" />);
    expect(screen.getByTestId("icon-eth")).toHaveAttribute("fill", "#ffffff");
  });

  it("is ImmutableX", () => {
    render(<CollectionTile {...sampleProps()} layer="Immutable" />);
    expect(screen.getByTestId("icon-im")).toHaveAttribute("fill", "#ffffff");
  });

  it("undefined layer", () => {
    render(<CollectionTile {...sampleProps()} layer={undefined} />);
    expect(screen.queryByRole("role")).toBeNull();
  });

  it("Tile can be clicked", () => {
    const onClick = jest.fn();
    render(<CollectionTile {...sampleProps()} onClick={onClick} />);
    userEvent.click(screen.getByRole("button"));
    expect(onClick).toBeCalled();
  });
});
