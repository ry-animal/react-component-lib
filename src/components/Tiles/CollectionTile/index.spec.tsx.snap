// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionTile /> Default loading matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  width: 80%;
  height: 25px;
  display: block;
  visibility: hidden;
  position: relative;
}

.c2 * {
  visibility: hidden;
  pointer-events: none;
  cursor: default;
}

.c2:after {
  content: "";
  visibility: visible;
  background-color: #EEEEEE;
  background-image: linear-gradient( 90deg, #EEEEEE, #EEEEEE, #F9F9F9, #EEEEEE, #EEEEEE );
  background-size: 200px 100%;
  background-repeat: no-repeat;
  border-radius: 8px;
  -webkit-animation: skeleton 1.3s ease-in-out infinite;
  animation: skeleton 1.3s ease-in-out infinite;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c3 {
  width: 50%;
  height: 25px;
  display: block;
  visibility: hidden;
  position: relative;
}

.c3 * {
  visibility: hidden;
  pointer-events: none;
  cursor: default;
}

.c3:after {
  content: "";
  visibility: visible;
  background-color: #EEEEEE;
  background-image: linear-gradient( 90deg, #EEEEEE, #EEEEEE, #F9F9F9, #EEEEEE, #EEEEEE );
  background-size: 200px 100%;
  background-repeat: no-repeat;
  border-radius: 8px;
  -webkit-animation: skeleton 1.3s ease-in-out infinite;
  animation: skeleton 1.3s ease-in-out infinite;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c6 {
  width: 40px;
  height: 40px;
  display: block;
  visibility: hidden;
  position: relative;
}

.c6 * {
  visibility: hidden;
  pointer-events: none;
  cursor: default;
}

.c6:after {
  content: "";
  visibility: visible;
  background-color: #EEEEEE;
  background-image: linear-gradient( 90deg, #EEEEEE, #EEEEEE, #F9F9F9, #EEEEEE, #EEEEEE );
  background-size: 200px 100%;
  background-repeat: no-repeat;
  border-radius: 40px;
  -webkit-animation: skeleton 1.3s ease-in-out infinite;
  animation: skeleton 1.3s ease-in-out infinite;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c8 {
  width: 40%;
  height: 10px;
  display: block;
  visibility: hidden;
  position: relative;
}

.c8 * {
  visibility: hidden;
  pointer-events: none;
  cursor: default;
}

.c8:after {
  content: "";
  visibility: visible;
  background-color: #EEEEEE;
  background-image: linear-gradient( 90deg, #EEEEEE, #EEEEEE, #F9F9F9, #EEEEEE, #EEEEEE );
  background-size: 200px 100%;
  background-repeat: no-repeat;
  border-radius: 8px;
  -webkit-animation: skeleton 1.3s ease-in-out infinite;
  animation: skeleton 1.3s ease-in-out infinite;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c9 {
  width: 60%;
  height: 10px;
  display: block;
  visibility: hidden;
  position: relative;
}

.c9 * {
  visibility: hidden;
  pointer-events: none;
  cursor: default;
}

.c9:after {
  content: "";
  visibility: visible;
  background-color: #EEEEEE;
  background-image: linear-gradient( 90deg, #EEEEEE, #EEEEEE, #F9F9F9, #EEEEEE, #EEEEEE );
  background-size: 200px 100%;
  background-repeat: no-repeat;
  border-radius: 8px;
  -webkit-animation: skeleton 1.3s ease-in-out infinite;
  animation: skeleton 1.3s ease-in-out infinite;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c10 {
  width: 25px;
  height: 25px;
  display: block;
  visibility: hidden;
  position: relative;
}

.c10 * {
  visibility: hidden;
  pointer-events: none;
  cursor: default;
}

.c10:after {
  content: "";
  visibility: visible;
  background-color: #EEEEEE;
  background-image: linear-gradient( 90deg, #EEEEEE, #EEEEEE, #F9F9F9, #EEEEEE, #EEEEEE );
  background-size: 200px 100%;
  background-repeat: no-repeat;
  border-radius: 25px;
  -webkit-animation: skeleton 1.3s ease-in-out infinite;
  animation: skeleton 1.3s ease-in-out infinite;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 475px;
  background-color: #FFFFFF;
  border-radius: 8px;
  padding: 16px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100px;
  padding: 16px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 25px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
}

.c5 {
  margin-right: 8px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
        height="25px"
        radius="8px"
        width="80%"
      />
      <span
        class="c3"
        height="25px"
        radius="8px"
        width="50%"
      />
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <span
          class="c6"
          height="40px"
          radius="40px"
          width="40px"
        />
      </div>
      <div
        class="c7"
      >
        <span
          class="c8"
          height="10px"
          radius="8px"
          width="40%"
        />
        <span
          class="c9"
          height="10px"
          radius="8px"
          width="60%"
        />
      </div>
      <div
        class="c5"
      >
        <span
          class="c10"
          height="25px"
          radius="25px"
          width="25px"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CollectionTile /> Default rendering matches snapshot (disabled) 1`] = `
<DocumentFragment>
  .c10 {
  background-image: url("creator.png");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 40px;
  width: 40px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c11 {
  border-radius: 50%;
}

.c9 {
  position: relative;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c9:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
}

.c8 {
  overflow: hidden;
  margin-right: 8px;
}

.c8 img {
  max-width: 100%;
}

.c12 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 400;
  color: #ffffff;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

.c13 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c14 {
  font-weight: 400;
  font-weight: 500;
}

.c2 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  background-color: rgba(0,0,0,0.3);
  border-radius: 8px;
  padding: 16px;
}

.c5 {
  padding: 12px;
  width: 100%;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #ffffff;
  font-size: 32px;
  font-weight: 600;
  margin: 8px 16px;
  line-height: 38px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  word-break: break-word;
}

.c3 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.c6 {
  width: calc(100% - 24px);
}

.c1 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  background: url(collection.jpg) center no-repeat #F9F9F9;
  background-size: cover;
}

.c0 {
  background-color: #000000;
  border-radius: 8px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 475px;
  overflow: hidden;
  pointer-events: none;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

@supports not (-webkit-backdrop-filter:blur(15px)) or (backdrop-filter:blur(15px)) {
  .c5 {
    background-color: rgba(0,0,0,0.3);
  }
}

@supports (-webkit-backdrop-filter:blur(15px)) or (backdrop-filter:blur(15px)) {
  .c5 {
    background-color: rgba(0,0,0,0.3);
    -webkit-backdrop-filter: blur(15px);
    backdrop-filter: blur(15px);
  }
}

<div
    aria-label="Collection name"
    class="c0"
    disabled=""
    role="button"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h5
          class="c4"
        >
          Collection name
        </h5>
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <div
                  class="c9"
                  size="40"
                >
                  <div
                    class="c10 c11"
                    size="40"
                  />
                </div>
              </div>
              <p
                class="c12"
              >
                <span
                  class="c13"
                >
                  Creator
                </span>
                <span
                  class="c13 c14"
                >
                  Creator
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CollectionTile /> Default rendering matches snapshot (size = small) 1`] = `
<DocumentFragment>
  .c4 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  background-color: rgba(0,0,0,0.3);
  border-radius: 8px;
  padding: 16px;
  border-radius: 4px;
  padding: 12px;
}

.c8 {
  color: #ffffff;
  font-size: 32px;
  font-weight: 600;
  margin: 8px 16px;
  line-height: 38px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  word-break: break-word;
}

.c6 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.c2 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  background: url(collection.jpg) center no-repeat #F9F9F9;
  background-size: cover;
}

.c0 {
  background-color: #000000;
  border-radius: 8px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 475px;
  overflow: hidden;
  pointer-events: none;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  height: auto;
  min-height: 120px;
  border-radius: 4px;
  cursor: pointer;
  pointer-events: auto;
}

.c0 .c7 {
  font-size: 12px;
  line-height: 12px;
  margin: 0;
}

.c0:hover,
.c0:focus {
  outline: none;
  border: none;
  box-shadow: 0px 2px 24px RGBA(0,0,0,0.24);
  -webkit-transform: translateY(-2.5px);
  -ms-transform: translateY(-2.5px);
  transform: translateY(-2.5px);
}

.c0:hover .c3,
.c0:focus .c3 {
  background-color: rgba(0,0,0,0.5);
}

.c0:hover .c5,
.c0:focus .c5 {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c0:hover .c1,
.c0:focus .c1 {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<div
    aria-label="Collection name"
    class="c0"
    role="button"
  >
    <div
      class="c1 c2"
    />
    <div
      class="c3 c4"
    >
      <div
        class="c5 c6"
      >
        <h5
          class="c7 c8"
        >
          Collection name
        </h5>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CollectionTile /> Default rendering matches snapshot 1`] = `
<DocumentFragment>
  .c13 {
  background-image: url("creator.png");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 40px;
  width: 40px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c14 {
  border-radius: 50%;
}

.c12 {
  position: relative;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c12:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
}

.c11 {
  overflow: hidden;
  margin-right: 8px;
}

.c11 img {
  max-width: 100%;
}

.c15 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 400;
  color: #ffffff;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

.c16 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c17 {
  font-weight: 400;
  font-weight: 500;
}

.c4 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  background-color: rgba(0,0,0,0.3);
  border-radius: 8px;
  padding: 16px;
}

.c8 {
  padding: 12px;
  width: 100%;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  color: #ffffff;
  font-size: 32px;
  font-weight: 600;
  margin: 8px 16px;
  line-height: 38px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  word-break: break-word;
}

.c6 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.c9 {
  width: calc(100% - 24px);
}

.c2 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  background: url(collection.jpg) center no-repeat #F9F9F9;
  background-size: cover;
}

.c0 {
  background-color: #000000;
  border-radius: 8px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 475px;
  overflow: hidden;
  pointer-events: none;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  cursor: pointer;
  pointer-events: auto;
}

.c0:hover,
.c0:focus {
  outline: none;
  border: none;
  box-shadow: 0px 2px 24px RGBA(0,0,0,0.24);
  -webkit-transform: translateY(-2.5px);
  -ms-transform: translateY(-2.5px);
  transform: translateY(-2.5px);
}

.c0:hover .c3,
.c0:focus .c3 {
  background-color: rgba(0,0,0,0.5);
}

.c0:hover .c5,
.c0:focus .c5 {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c0:hover .c1,
.c0:focus .c1 {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

@supports not (-webkit-backdrop-filter:blur(15px)) or (backdrop-filter:blur(15px)) {
  .c8 {
    background-color: rgba(0,0,0,0.3);
  }
}

@supports (-webkit-backdrop-filter:blur(15px)) or (backdrop-filter:blur(15px)) {
  .c8 {
    background-color: rgba(0,0,0,0.3);
    -webkit-backdrop-filter: blur(15px);
    backdrop-filter: blur(15px);
  }
}

<div
    aria-label="Collection name"
    class="c0"
    role="button"
  >
    <div
      class="c1 c2"
    />
    <div
      class="c3 c4"
    >
      <div
        class="c5 c6"
      >
        <h5
          class="c7"
        >
          Collection name
        </h5>
        <div
          class="c8"
        >
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              <div
                class="c11"
              >
                <div
                  class="c12"
                  size="40"
                >
                  <div
                    class="c13 c14"
                    size="40"
                  />
                </div>
              </div>
              <p
                class="c15"
              >
                <span
                  class="c16"
                >
                  Creator
                </span>
                <span
                  class="c16 c17"
                >
                  Creator
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
